### PR TITLE
Improve API Docs

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**
## Purpose
$Subject

## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,7 +9,9 @@ Ballerina Regex Library
 
 This Regex library is one of the standard libraries of the <a target="_blank" href="https://ballerina.io/">Ballerina</a> language.
 
-It has support for regex functionalities like checking whether a string matches a given regex, replacing substrings, and splitting strings based on a regex.
+This provides functions for regex functionalities like checking whether a string matches a given regex, replacing substrings, and splitting strings based on a regex.
+
+For example demonstrations of the usage, go to [Ballerina By Examples](https://ballerina.io/learn/by-example/).
```
Comment (daneshk) [inline README.md:14]:
```suggestion
```

File: regex-ballerina/Module.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+## Module Overview
```
Comment (daneshk) [inline regex-ballerina/Module.md:1]:
```suggestion
## Overview
```

File: regex-ballerina/natives.bal [hunk 3]
```diff
@@ -17,7 +17,7 @@
 import ballerina/jballerina.java;
 
 
-# Checks whether the given string matches the provided regex.
+# Checks whether the given string matches the provided `regex`.
```
Comment (daneshk) [inline regex-ballerina/natives.bal:20]:
```suggestion
# Checks whether the given string matches the provided `regex` expression.
```

Comment (daneshk) [inline regex-ballerina/natives.bal:20]: In some places we used `regular expressions` and in some places, we use `regex`. can you check and make them consistent

File: regex-ballerina/Module.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+
+This module provides functions for searching, splitting, and replacing the set of characters of the string. These APIs are using a 
+regular expression as a `String` to perform these operations by finding matches string.
+
+[Regular expressions](https://en.wikipedia.org/wiki/Regular_expression), are a notation for describing sets of
```
Comment (MadhukaHarith92) [inline regex-ballerina/Module.md:6]:
```suggestion
[Regular expressions](https://en.wikipedia.org/wiki/Regular_expression), are notations for describing sets of 
```

File: regex-ballerina/Package.md [hunk 5]
```diff
@@ -1,8 +1,15 @@
-## Package overview
+## Package Overview
 
-This package provides the following regex utility functions. 
+This Regex library is one of the standard libraries of the <a target="_blank" href="https://ballerina.io/">Ballerina</a> language.
```
Comment (MadhukaHarith92) [inline regex-ballerina/Package.md:3]:
```suggestion
The Regex library is one of the standard libraries of the <a target="_blank" href="https://ballerina.io/">Ballerina</a> language.
```

File: README.md (hunk 1 above)
Comment (kalaiyarasiganeshalingam) [reply README.md:14]: Do we need to remove this from README file too?

File: regex-ballerina/natives.bal (hunk 3 above)
Comment (kalaiyarasiganeshalingam) [reply regex-ballerina/natives.bal:20]: Changed to regex

File: README.md [hunk 6]
```diff
@@ -9,7 +9,11 @@ Ballerina Regex Library
 
 This Regex library is one of the standard libraries of the <a target="_blank" href="https://ballerina.io/">Ballerina</a> language.
 
-It has support for regex functionalities like checking whether a string matches a given regex, replacing substrings, and splitting strings based on a regex.
+This provides functions for regex functionalities like checking whether a string matches a given regex, replacing substrings, and splitting strings based on a regex.
```
Comment (praneesha) [inline README.md:12]:
```suggestion
This provides functions for regex functionalities such as checking whether a string matches a given regex, replacing substrings, and splitting strings based on a regex.
```

File: regex-ballerina/Module.md [hunk 7]
```diff
@@ -0,0 +1,9 @@
+## Overview
+
+This module provides functions for searching, splitting, and replacing the set of characters of the string. These APIs are using a 
+regular expression as a `String` to perform these operations by finding matches string.
```
Comment (praneesha) [inline regex-ballerina/Module.md:4]:
```suggestion
regular expression as a `String` to perform these operations by finding the string matches.
```

File: regex-ballerina/Package.md [hunk 8]
```diff
@@ -1,8 +1,15 @@
...
+## Package Overview
 
-This package provides the following regex utility functions. 
+The Regex library is one of the standard libraries of the <a target="_blank" href="https://ballerina.io/">Ballerina</a> language.
 
-* `matches` - Checks whether a string matches a given regex.
-* `replaceAll` - Replaces every occurrence of a substring that matches a given regex in a string, with another string provided.
-* `replaceFirst` - Replaces the first occurrence of a substring that matches a given regex in a string, with another string provided.
-* `split` - Splits a string using a given regex delimiter and returns an array of the split elements.                     
+This package provides functions for regex functionalities like checking whether a string matches a given regex, replacing substrings, and splitting strings based on a regex.
```
Comment (kalaiyarasiganeshalingam) [inline regex-ballerina/Package.md:5]:
```suggestion
This package provides functions for regex functionalities such as checking whether a string matches a given regex, replacing substrings, and splitting strings based on a regex.
```

File: README.md [hunk 9]
```diff
@@ -9,7 +9,11 @@ Ballerina Regex Library
 
 This Regex library is one of the standard libraries of the <a target="_blank" href="https://ballerina.io/">Ballerina</a> language.
```
Comment (kalaiyarasiganeshalingam) [inline README.md:10]:
```suggestion
The `regex` library is one of the standard libraries of the <a target="_blank" href="https://ballerina.io/">Ballerina</a> language.
```

File: regex-ballerina/Package.md [hunk 10]
```diff
@@ -1,8 +1,15 @@
-## Package overview
+## Package Overview
 
-This package provides the following regex utility functions. 
+The Regex library is one of the standard libraries of the <a target="_blank" href="https://ballerina.io/">Ballerina</a> language.
```
Comment (kalaiyarasiganeshalingam) [inline regex-ballerina/Package.md:3]:
```suggestion
The `regex` library is one of the standard libraries of the <a target="_blank" href="https://ballerina.io/">Ballerina</a> language.
```

**Review by praneesha — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (daneshk) [reply README.md:16]: No. I thought it is module.MD

**Review by daneshk — APPROVED** [review]
LGTM

**Pull request merged**